libraries/Microsoft.Bot.Builder.AI.QnA/Dialogs/QnAMakerDialog.cs

```diff
@@ -504,7 +474,7 @@ protected virtual async Task<QnADialogResponseOptions> GetQnAResponseOptionsAsyn
         {
             return new QnADialogResponseOptions
             {
-                NoAnswer = await this.NoAnswer.BindAsync(dc, dc.State).ConfigureAwait(false),
+                NoAnswer = await this.NoAnswer.BindAsync(dc, dc.ActiveDialog.State).ConfigureAwait(false),
```

Comment:
It looks like NoAnswer can be made just an Activity and we can remove Microsoft.Bot.Builder.Dialogs.ITemplate<T> and Microsoft.Bot.Builder.AI.QnA.Utils.BindToActivity